native/src/routes/NewsContainer.tsx

```diff
@@ -1,319 +1,86 @@
-import createNavigate from '../navigation/createNavigate'
-import { NewsModelsType, StateType } from '../redux/StateType'
-import { FetchMoreNewsActionType, StoreActionType } from '../redux/StoreActionType'
-import { reportError } from '../utils/sentry'
-import News from './News'
+import useCities from '../hooks/useCities'
+import LocalNews from './LocalNews'
+import TuNews from './TuNews'
 
 type NavigationPropsType = {
   route: RoutePropType<NewsRouteType>
   navigation: NavigationPropType<NewsRouteType>
 }
-type OwnPropsType = NavigationPropsType
-type DispatchPropsType = {
-  dispatch: Dispatch<StoreActionType>
-}
-type ContainerPropsType =
-  | (NavigationPropsType &
-      DispatchPropsType & {
-        status: 'fetching'
-        newsId: string | null | undefined
-        language: string
-        cityModel: CityModel
-        selectedNewsType: NewsType
-        page: null
-        news: null
-        hasMoreNews: null
-        isFetchingMore: null
-      })
-  | (NavigationPropsType &
-      DispatchPropsType & {
-        status: 'ready'
-        news: NewsModelsType
-        hasMoreNews: boolean | null | undefined
-        page: number | null
-        isFetchingMore: boolean
-        newsId: string | null | undefined
-        language: string
-        cityModel: CityModel
-        selectedNewsType: NewsType
-      })
-type RefreshPropsType = NavigationPropsType & {
-  cityCode: string
-  language: string
-  newsId: string | null | undefined
-  selectedNewsType: NewsType
-}
-type StatePropsType = StatusPropsType<ContainerPropsType, RefreshPropsType>
 
-const refresh = (refreshProps: RefreshPropsType, dispatch: Dispatch<StoreActionType>) => {
-  const { route, navigation, cityCode, language, newsId, selectedNewsType } = refreshProps
-  const navigateTo = createNavigate(dispatch, navigation)
-  navigateTo(
-    {
-      route: NEWS_ROUTE,
-      cityCode,
-      newsType: selectedNewsType,
-      languageCode: language,
-      newsId: newsId || undefined
-    },
-    route.key,
-    true
+const NewsContainer = ({ route: { params }, navigation }: NavigationPropsType): ReactElement => {
+  const { cityCode, languageCode, newsType, newsId } = params
+  const [selectedNewsType, setSelectedNewsType] = useState<NewsType>(newsType)
+  const [selectedNewsId, setSelectedNewsId] = useState<string | null>(newsId)
+  const [selectedLanguage, setSelectedLanguage] = useState<string>(languageCode)
+  const cities = useCities()
+  const dispatch = useDispatch()
+  const isLocalNews = selectedNewsType === LOCAL_NEWS_TYPE
+
+  const cityModel = cities?.find(model => model.code === cityCode)
+
+  useEffect(
+    // Handle back navigation: If we are at a news detail screen navigate back to overview instead of closing the route
+    () =>
+      navigation.addListener('beforeRemove', e => {
+        if (selectedNewsId) {
+          e.preventDefault()
+          setSelectedNewsId(null)
+        }
+      }),
```

Comment:
Should the Listener be removed?

Reply:
What do you mean by that? Like completely or just returning the remove function to make sure everything is cleaned up correctly?
If the latter, it actually is already as `navigation.addListener` returns the remove function.

Reply:
yeah i meant the cleanup. overlooked it.